challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -10506,6 +10506,1096 @@
+        }
+      ],
+      "id": "5a23c84252665b21eecc7f39",
+      "challengeType": 5,
+      "releasedOn": "July 28, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": [
+            "function Nonoblock (data, len) {",
+            "  // Good luck!",
+            "}"
+          ],
+          "head": [],
+          "tail": [
+            "var results=[",
+            "[\"11010\",\"11001\",\"01101\"],",
+            "[\"00000\"],",
+            "[\"1111111100\",\"0111111110\",\"0011111111\"],",
+            "[\"110111011011100\",\"110111011001110\",\"110111011000111\",\"110111001101110\",\"110111001100111\",\"110111000110111\",\"110011101101110\",\"110011101100111\",\"110011100110111\",\"110001110110111\",\"011011101101110\",\"011011101100111\",\"011011100110111\",\"011001110110111\",\"001101110110111\"],",
+            "[]]"
+          ]
+        }
+      }
+    },
+    {
+      "title": "Nonogram solver",
+      "description": [
+        "A <a href=\"https://en.wikipedia.org/wiki/Nonogram\">nonogram</a> is a puzzle that provides numeric clues used to fill in a grid of cells, establishing for each cell whether it is filled or not.",
+        "The puzzle solution is typically a picture of some kind.",
+        "Each row and column of a rectangular grid is annotated with the lengths of its distinct runs of occupied cells.",
+        "Using only these lengths you should find a valid configuration of empty and occupied cells.",
+        "Problem:",
+        ". . . . . . . .  3",
+        ". . . . . . . .  2 1",
+        ". . . . . . . .  3 2",
+        ". . . . . . . .  2 2",
+        ". . . . . . . .  6",
+        ". . . . . . . .  1 5",
+        ". . . . . . . .  6",
+        ". . . . . . . .  1",
+        ". . . . . . . .  2",
+        "1 3 1 7 5 3 4 3",
+        "2 1 5 1",
+        "Solution:",
+        ". # # # . . . .  3",
+        "# # . # . . . .  2 1",
+        ". # # # . . # #  3 2",
+        ". . # # . . # #  2 2",
+        ". . # # # # # #  6",
+        "# . # # # # # .  1 5",
+        "# # # # # # . .  6",
+        ". . . . # . . .  1",
+        ". . . # # . . .  2",
+        "1 3 1 7 5 3 4 3",
+        "2 1 5 1",
+        "The problem above could be represented by two lists of lists:",
+        "x = [[3], [2,1], [3,2], [2,2], [6], [1,5], [6], [1], [2]]",
+        "y = [[1,2], [3,1], [1,5], [7,1], [5], [3], [4], [3]]",
+        "A more compact representation of the same problem uses strings,",
+        "where the letters represent the numbers, A=1, B=2, etc:",
+        "x = \"C BA CB BB F AE F A B\"",
+        "y = \"AB CA AE GA E C D C\"",
+        "Write a function that takes an array of strings as parameter and returns a 2D bit-matrix as a solution for the given nonogram (represented by the array of strings). The element of the bit-matrix should be 1 for occupied cell and 0 for unoccupied cell."
```

Comment:
"Write a function that takes an array of strings as parameter and returns a 2D bit-matrix as a solution for the given nonogram (represented by the array of strings). Each element of the bit-matrix should be either a 1 for an occupied cell, or a 0 for an unoccupied cell."